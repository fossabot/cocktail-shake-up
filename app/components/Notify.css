.notification {
  position: fixed;
  bottom: 0;

  width: 100%;
  height: 0;

  transition: height 0.25s, opacity 0.5s;
  text-align: center;

  opacity: 0;
  color: white;
  background-color: grey;
}

.notification.visible {
  height: auto;
  min-height: 8rem;
  padding: 1rem 0 1rem 0;

  transition: height 0.25s, opacity 0.5s;

  opacity: 1;
}

.notification .close {
  float: right;

  width: 2.4rem;
  height: 2rem;
  margin: 0 1rem;

  color: darkred;
}

.notification h1 {
  color: transparent;
}

.notification.visible a {
  color: white;
}

.notification.visible h1 {
  margin: 0 0 1rem;

  color: white;
}
