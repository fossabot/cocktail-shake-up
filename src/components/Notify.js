/* @flow */

import React from 'react';
import './Notify.css';

const Notify = (props: Object) => {
  return (
    <div className={props.visible ? 'notification visible' : 'notification'}>
      {props.onCloseHandler ? <button className="close" onClick={props.onCloseHandler} /> : null}
      <h1>{props.title}</h1>
      {props.body}
    </div>
  );
};

export default Notify;
