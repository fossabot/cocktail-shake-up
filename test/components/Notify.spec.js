import React from 'react';
import Notify from '../../src/components/Notify';
import { shallow } from 'enzyme';

test('Should render a notification', () => {
  const tree = shallow(<Notify />).debug();
  expect(tree).toMatchSnapshot();
});

test('Should render a notification with a closeHandler', () => {
  const tree = shallow(<Notify onCloseHandler={() => {}} />).debug();
  expect(tree).toMatchSnapshot();
});

test('should render a notification with a title', () => {
  const tree = shallow(<Notify visible={true} />).debug();
  expect(tree).toMatchSnapshot();
});
