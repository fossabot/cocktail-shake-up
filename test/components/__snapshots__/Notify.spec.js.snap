// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a Notification 1`] = `
"<div className=\\"notification\\">
<h1 />
</div>"
`;

exports[`Should render a Notification with a closeHandler 1`] = `
"<div className=\\"notification\\">
<button className=\\"close\\" onClick={[Function]} />
<h1 />
</div>"
`;

exports[`should render a Notification with a title 1`] = `
"<div className=\\"notification visible\\">
<h1 />
</div>"
`;
